### ISSUE TITLE
IndexedDB database crate 

### ISSUE THREAD

**Issue by justinmoon** [fedimint/fedimint#841]
@Maan2003 wrote an initial database implementation a couple months ago. It's currently used by [webimint](https://github.com/justinmoon/webimint/blob/master/webimint/src/db.rs). It would be nice if this implementation was upstreamed to this repo so that it's maintained more diligently.

**Comment by m1sterc001guy** [comment]
Now that we have SQLite support I don't think we need to upstream Rexie. Should we wait to close until we've tested SQLite in WASM?

**Comment by justinmoon** [comment]
Made a new issue for SQLite in WASM. Let's leave this open until we resolve that.

**Comment by elsirion** [comment]
Webimint has a hacky version of this: https://github.com/elsirion/webimint-rs/blob/master/src/db.rs

Just needs to be extracted into a crate.

**Comment by tvolk131** [comment]
@elsirion is your [webkv](https://github.com/elsirion/webkv) project relevant here?

**Comment by elsirion** [comment]
Yes, that's exactly the idea behind `webkv`. [I already have a Fedimint DB impl using it](https://github.com/elsirion/fedimint-webkv), but needs more testing and docs. I experienced a deadlock in webimint, unclear if caused by `webkv` or `fedimint-client` and it's hard to debug on the web, need to get back to it.
